src/builder.ts

```diff
@@ -1297,6 +1313,22 @@ export class GatewayBuilder {
 
         break;
 
+      case SupportedProtocols.Lido:
+          this._metadata.vault = await lido.infos.getVault(
+            this._provider.connection,
+            withdrawParams.vaultId
+          );
+          this.params.validatorIndex = withdrawParams.validatorIndex;
```

Comment:
Perhaps add some descriptive comment here to explain the intention? Since this is a temporary work-around